src/metabase/data_app/api/data_app.clj

```diff
@@ -0,0 +1,108 @@
+(ns metabase.data-app.api.data-app
+  "/api/data-app endpoints for data apps CRUD operations"
+  (:require
+   [honey.sql.helpers :as sql.helpers]
+   [metabase.api.common :as api]
+   [metabase.api.macros :as api.macros]
+   [metabase.data-app.models :as data-app.models]
+   [metabase.request.core :as request]
+   [metabase.util :as u]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(defn- get-data-app
+  "Get a data app by id with latest definitions and release info."
+  [id]
+  (let [data-app (api/read-check (t2/select-one :model/DataApp id))]
+    (assoc data-app
+           :latest_definition (data-app.models/latest-definition id)
+           :released_definition (data-app.models/released-definition id)
+           :latest_release (data-app.models/latest-release id))))
+
+(defn- data-app-clauses
+  "Honeysql clauses for filtering data apps with status and pagination"
+  [status limit offset]
+  (cond-> {}
+    (some? status) (sql.helpers/where [:= :status (keyword status)])
+    (some? limit) (sql.helpers/limit limit)
+    (some? offset) (sql.helpers/offset offset)))
+
+(defn- filter-clauses-without-paging
+  "Given a where clause, return a clause that can be used to count."
+  [clauses]
+  (dissoc clauses :order-by :limit :offset))
+
+(api.macros/defendpoint :get "/"
+  "List all data apps. Optionally filter by status.
+
+  Takes `limit`, `offset` for pagination."
+  [_route-params
+   {:keys [status]} :- [:map
+                        [:status {:optional true} [:maybe [:enum "private" "published" "archived"]]]]]
+  (let [limit   (request/limit)
+        offset  (request/offset)
+        clauses (data-app-clauses status limit offset)]
+    {:data (t2/select :model/DataApp
+                      (sql.helpers/order-by clauses
+                                            [:created_at :desc]
+                                            [:id :asc]))
+     :total (t2/count :model/DataApp (filter-clauses-without-paging clauses))
+     :limit limit
+     :offset offset}))
```

Comment:
It makes me sad that we (Metabase) do pagination "wrong" like this, rather than using "before" and "after" inequalities which are faster and avoid skipping over new data, but I guess it makes sense to just be consistent.